packages/client/lib/Swap.js

```diff
@@ -42,6 +43,11 @@ export default class Swap {
    * @return {Promise<string>} Resolves with a 32 byte secret
    */
   async generateSecret (message) {
+    try {
+      return this.client.getMethod('generateSecret')(message)
```

Comment:
Are we implementing `generateSecret` anywhere else?

Reply:
Yes. `BitcoinNodeWalletProvider` has its own implementation because for node wallet, it is only possible to sign messages using a legacy address. `getAddresses` call returns addresses based on the type of address pre-specified in constructor (could be bech32/p2sh addresses) therefore for secret generation to continue working under these conditions, the node wallet needs to do it differently.